File: solution/scala/day01/src/main/scala/Food.scala
import java.time.LocalDate
import java.util.UUID

case class Food(
    expirationDate: LocalDate,
    approvedForConsumption: Boolean,
    insperctorId: Option[UUID]
) {
  def isEdible(now: LocalDate): Boolean =
    isFresh(now) &&
      hasBeenInspected &&
      canBeConsumed

  private def isFresh(now: LocalDate): Boolean = {
    expirationDate.isAfter(now)
  }

  private def hasBeenInspected: Boolean = {
    insperctorId.nonEmpty
  }

  private def canBeConsumed: Boolean = {
    approvedForConsumption
  }

}

File: solution/rust/day01/src/food.rs
use chrono::{DateTime, Utc};

pub struct Food {
    pub expiration_date: DateTime<Utc>,
    pub approved_for_consumption: bool,
    pub inspector_id: Option<uuid::Uuid>,
}

impl Food {
    pub fn is_edible(&self, now: &DateTime<Utc>) -> bool {
        self.is_fresh(now) &&
            self.can_be_consumed() &&
            self.has_been_inspected()
    }

    fn is_fresh(&self, now: &DateTime<Utc>) -> bool {
        self.expiration_date > *now
    }

    fn has_been_inspected(&self) -> bool {
        self.inspector_id.is_some()
    }

    fn can_be_consumed(&self) -> bool {
        self.approved_for_consumption
    }

}